neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/Neo4jDialect.java

```diff
@@ -6,592 +6,17 @@
-import org.hibernate.ogm.dialect.query.spi.BackendQuery;
-import org.hibernate.ogm.dialect.query.spi.ClosableIterator;
-import org.hibernate.ogm.dialect.query.spi.QueryParameters;
-import org.hibernate.ogm.dialect.spi.AssociationContext;
-import org.hibernate.ogm.dialect.spi.ModelConsumer;
-import org.hibernate.ogm.dialect.spi.NextValueRequest;
-import org.hibernate.ogm.dialect.spi.TupleAlreadyExistsException;
-import org.hibernate.ogm.dialect.spi.TupleContext;
-import org.hibernate.ogm.model.key.spi.AssociatedEntityKeyMetadata;
-import org.hibernate.ogm.model.key.spi.AssociationKey;
-import org.hibernate.ogm.model.key.spi.AssociationKeyMetadata;
-import org.hibernate.ogm.model.key.spi.EntityKey;
-import org.hibernate.ogm.model.key.spi.EntityKeyMetadata;
-import org.hibernate.ogm.model.key.spi.RowKey;
-import org.hibernate.ogm.model.spi.Association;
-import org.hibernate.ogm.model.spi.AssociationOperation;
-import org.hibernate.ogm.model.spi.EntityMetadataInformation;
-import org.hibernate.ogm.model.spi.Tuple;
-import org.hibernate.ogm.model.spi.TupleOperation;
-import org.hibernate.ogm.persister.impl.OgmCollectionPersister;
-import org.hibernate.ogm.persister.impl.OgmEntityPersister;
-import org.hibernate.persister.collection.CollectionPersister;
-import org.hibernate.persister.entity.EntityPersister;
-import org.neo4j.graphdb.ConstraintViolationException;
-import org.neo4j.graphdb.Direction;
-import org.neo4j.graphdb.GraphDatabaseService;
-import org.neo4j.graphdb.Node;
-import org.neo4j.graphdb.QueryExecutionException;
-import org.neo4j.graphdb.Relationship;
-import org.neo4j.graphdb.ResourceIterator;
-import org.neo4j.graphdb.Result;
-import org.neo4j.kernel.api.exceptions.schema.UniquePropertyConstraintViolationKernelException;
 
 /**
- * Abstracts Hibernate OGM from Neo4j.
- * <p>
- * A {@link Tuple} is saved as a {@link Node} where the columns are converted into properties of the node.<br>
- * An {@link Association} is converted into a {@link Relationship} identified by the {@link AssociationKey} and the
- * {@link RowKey}. The type of the relationship is the value returned by
- * {@link AssociationKeyMetadata#getCollectionRole()}.
- * <p>
- * If the value of a property is set to null the property will be removed (Neo4j does not allow to store null values).
+ * Equivalent to {@link EmbeddedNeo4jDialect}.
  *
- * @author Davide D'Alto &lt;davide@hibernate.org&gt;
+ * @see EmbeddedNeo4jDialect
+ * @author Davide D'Alto
  */
-public class Neo4jDialect extends BaseNeo4jDialect {
-
-	private static final Log log = LoggerFactory.getLogger();
-
-	private final GraphDatabaseService dataBase;
-
-	private final EmbeddedNeo4jSequenceGenerator sequenceGenerator;
-
-	private Map<EntityKeyMetadata, EmbeddedNeo4jEntityQueries> entityQueries;
-
-	private Map<AssociationKeyMetadata, EmbeddedNeo4jAssociationQueries> associationQueries;
+public class Neo4jDialect extends EmbeddedNeo4jDialect {
```

Comment:
Is there really a point in keeping this dialect around?

Personally, I think we should simply remove it: we will have prominent information about the support of the Remote protocol in the next release so I think it's the best moment to remove it and explain to people they have to move to the Embedded one or consider using the Remote one.

If we deprecate it, they probably won't notice it and be surprised when we will remove it as we probably won't have a major change in the Neo4j dialects to advertise.

I'm quite new with this sort of things so I might not be in line with the usual way of doing things.

And I think we should do the same for AvailableDatastoreProvider.NEO4J. IMHO, it's kinda easy to explain we now have 2 Neo4j dialects and people have to choose between the two. If they want to keep the same behavior as before, they have to switch to the NEOJ_EMBEDDED one.

If we don't remove it now, please consider adding back the @Deprecated annotation and state that it will be removed in the next release. I really think we shouldn't keep these old things for too long: it's quite confusing for new users.